src/Native/Bootstrap/CppCodeGen.h

```diff
@@ -29,6 +33,14 @@
 #define CORERT_UNREACHABLE  __builtin_unreachable()
 #endif
 
+#if defined(_MSC_VER)
+#define CORERT_DEBUGBREAK __debugbreak()
+#elif defined(__unix__) || defined(__APPLE__)
+#define CORERT_DEBUGBREAK raise(SIGTRAP)
```

Comment:
`__builtin_trap()` may be better than taking a dependency on signals here.

Reply:
I thought `__builtin_trap` assumes the execution irrecoverably stops after it's reached. E.g. [here](https://stackoverflow.com/questions/173618/is-there-a-portable-equivalent-to-debugbreak-debugbreak) someone says they get a SIGILL from it.

Reply:
Good point.

I have notice that we have `RhDebugBreak` in the PAL. Should we use that instead? And fix it to use `raise(SIGTRAP)` instead of unconditional abort - can be done separately.

My meta concern is about not including more stuff into this file because of these includes tend to collide in surprising ways with the generated C++.

Reply:
> I have notice that we have `RhDebugBreak` in the PAL

Nice find. Fixed to use that instead.